pkg/process/cache.go

```diff
@@ -109,14 +117,28 @@ func (pc *Cache) deletePending(process *ProcessInternal) {
 	pc.deleteChan <- process
 }
 
-func (pc *Cache) refDec(p *ProcessInternal) {
+func (pc *Cache) refDec(p *ProcessInternal, reason string) {
+	p.refcntOpsLock.Lock()
+	if val, ok := p.refcntOps[reason]; ok {
+		p.refcntOps[reason] = val + 1
+	} else {
+		p.refcntOps[reason] = 1
+	}
+	p.refcntOpsLock.Unlock()
 	ref := atomic.AddUint32(&p.refcnt, ^uint32(0))
 	if ref == 0 {
 		pc.deletePending(p)
 	}
 }
 
-func (pc *Cache) refInc(p *ProcessInternal) {
+func (pc *Cache) refInc(p *ProcessInternal, reason string) {
+	p.refcntOpsLock.Lock()
+	if val, ok := p.refcntOps[reason]; ok {
+		p.refcntOps[reason] = val + 1
+	} else {
+		p.refcntOps[reason] = 1
+	}
```

Comment:
I think you can simplify this by:
```suggestion
	p.refcntOps[reason]++
```
since Golang initializes things at 0 anyway.

same for decrement, I would add a comment on decrement on why `++` is okay and not a bug since Djalal was confused about this at first read and me too :D!

Reply:
Yes, makes sense. I have also added a comment as you asked for.